spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1745,12 +1783,13 @@ public Boolean isMandatoryFor(final Message message) {
 			}
 		}
 		else {
-			return doExecute(action, connectionFactory);
+			return doExecute(action, connectionFactory, publisherConnectionIfPossible);
 		}
 	}
 
 
-	private <T> T doExecute(ChannelCallback<T> action, ConnectionFactory connectionFactory) {
+	private <T> T doExecute(ChannelCallback<T> action, ConnectionFactory connectionFactory,
+			boolean publisherConnectionIfPossible) {
```

Comment:
Why can't we just rely on the `usePublisherConnection` property? Why do we need the method param context?

Reply:
The rationale I used was to use the publisher channel only for `send()` operations, not `receive()` operations.

Now I remember why I added an overloaded `execute()` operation with the boolean, so the user can choose.

Reply:
Well, why?

The current solution and recommendation is to use a separate CCF for the `RabbitTemplate` to avoid dead-lock on the consumer side. This way any template operations are performed on the same injected CF. So, I would expect here consistency as well. Right, I understand that these `receive()` operations are not publishing per se, but does it matter in the end? We have a dead lock in the consumer when producer is blocked on the same connection. But in these `receive()` operations we don't have any producing functionality and in most cases they are in the beginning of the logic. Therefore, IMO, we are good to perform them in any connection and from here our API would be still simple.

WDYT?

Reply:
OK; you've worn me down.